### ISSUE TITLE
broken

### ISSUE THREAD

**Issue by stevenvachon** [gorangajic/isbot#4]
The previous version:
```js
isbot("googleBot asdf") //=> googleBot
isbot("asdfasdf") //=> false
```
vs current version:
```js
isbot("googleBot asdf") //=> Bot
isbot("asdfasdf") //=> false
```

It's not useful like this.

**Comment by stevenvachon** [comment]
@gorangajic ?

**Comment by gorangajic** [comment]
What is the difference? Point of this module is just to provide a way to tell if user agent string is bot or not. 

I forget to answer last time, sorry about that.

**Comment by stevenvachon** [comment]
Because I was using the result in the previous version to store `X-Robots-Tag`-compatible user-agent keys:
```js
var userAgent = "asdf asdf Googlebot/2.1 asdf";
userAgent = isbot(userAgent.toLowerCase()); //=> googlebot
if (userAgent===false) userAgent = "robots";
var html = '<meta name="'+userAgent+'" value="noindex">'
```
More info: https://github.com/stevenvachon/robot-directives

**Comment by stevenvachon** [comment]
Additionally, your documentation suggests functionality that no longer works.

**Comment by gorangajic** [comment]
there is no Googlebot anymore in bot list, that's why it's not there anymore

**Comment by gorangajic** [comment]
https://github.com/gorangajic/isbot/commit/46307776492837273ebb72ac60a131267a5a1d0c

**Comment by gorangajic** [comment]
Published @1.0.1

**Comment by timbowhite** [comment]
Hey just a few thoughts here.

The 'Googlebot' entries were removed in the bot list in v1.0.0 because this was assumed:

> Point of this module is just to provide a way to tell if user agent string is bot or not. 

They were removed for performance reasons: the smaller the list, the faster the performance of the matching regex.  As the term 'bot' is already in the generic portion of the list, there's no need to repeat terms containing that substring.

If the point of this module is to return matching bot string "tokens" then I'd suggest that intent be made clear in the README.md and tests be written to ensure that expected substring tokens are returned for bot user agent strings.

(Personally, I think the expected return for is* named functions is a boolean.)

**Comment by stevenvachon** [comment]
@timbowhite 3 array indexes is a micro-optimization.

Perhaps `isbot()` should be a boolean and a sub-function of `isbot.name()` returns the token.

**Comment by gorangajic** [comment]
I agree with you @timbowhite initial and main thing of this module is to return if user agent is bot, so we should just return boolean when useragent is found

sorry @stevenvachon but you can fork code and publish your own module with different name or just use old version. 

Will publish tomorrow version that just return boolean

**Comment by timbowhite** [comment]
> @timbowhite 3 array indexes is a micro-optimization.

Maybe but I'm pretty sure I removed > 3 ua specific matching strings from the list.js because they were redundant, [take a look here](https://github.com/gorangajic/isbot/commit/c3181e2da58d5b7ee36c19711f74d9cef909df3c).

@gorangajic sounds good.  If you're switching to a boolean return, might I also suggest removing the parenthesis from the regex and using ```test``` instead of ```match```, [seems like it's the fastest regex method](https://jsperf.com/exec-vs-match-vs-test-vs-search/2) for basic true/false testing.

**Comment by stevenvachon** [comment]
@gorangajic I'd prefer to do neither. What is so bad about supplying an alternative function? It'd be like 3 lines of additional code.

**Comment by stevenvachon** [comment]
After further research, this package can be combined with [useragent](https://npmjs.com/useragent):
```js
var useragent = require("useragent");
var isbot = require("isbot");

var test = useragent("Mozilla/5.0 Googlebot/2.1 (+http://www.googlebot.com/bot.html)");
test = test.family.toLowerCase();

if (test!=="other" && isbot(test)===true) {
    console.log(test); //=> googlebot
}
```

This is more accurate anyway.

With this in mind, you could even build your lib this way and reduce the length of your search list.

**Comment by gorangajic** [comment]
Great then you can go with that module to figure out bot type.

I have published 2.0.0 that always returns boolean
